flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSink.java

```diff
@@ -192,6 +196,9 @@ public CommonExecSink(
                 finalInputTransform = new PartitionTransformation<>(inputTransform, partitioner);
                 finalInputTransform.setParallelism(parallelism);
             }
+
+            final SinkOperator operator =
+                    new SinkOperator(env.clean(sinkFunction), rowtimeFieldIndex, enforcer);
```

Comment:
I think we need to do some refactoring for the `createSinkTransformation` method, currently the logic is quite mess. One idea is we can make enforcer always as an single operator if there are not null fields, so the enforcer will not be wrap in the `SinkOperator`.